src/services/dataRenderService.ts

```diff
@@ -558,12 +586,15 @@ export class DataRenderService {
             context.append("g").classed(DataRenderService.labelGraphicsContextClass.className, true);
         }
 
-        let labels: Selection<any> = context
-            .select(DataRenderService.labelGraphicsContextClass.selectorName)
-            .selectAll(DataRenderService.DataLabels.selectorName)
-            .data<LabelEnabledDataPoint>(
+        let labels = context
+            .select<SVGGElement>(DataRenderService.labelGraphicsContextClass.selectorName)
+            .selectAll<SVGTextElement, d3PieArcDatum<AsterDataPoint>>(DataRenderService.DataLabels.selectorName)
+            .data(
                 filteredData,
-                (d: AsterArcDescriptor) => (<ISelectionId>d.data.identity).getKey());
+                (d: d3PieArcDatum<AsterDataPoint>) => {
+                    // TODO: check the types
```

Comment:
Is this TODO actual?

Reply:
No, the TODO is unnecessary as the actual type is already written.